External/FEXCore/Source/Interface/Core/Dispatcher/X86Dispatcher.cpp

```diff
@@ -25,384 +26,338 @@
+    HostCode = X86JITCore::ExitFunctionLink(Frame,record);
+    sigprocmask( SIG_SETMASK, &ProcMask, NULL);
+  }
+  
+  return HostCode;
+}
 
-      mov(rdi, SIG_SETMASK);
-      mov(rsi, rsp);
-      mov(rdx, 0); // Don't care about result
-      mov(r10, 8);
-      mov(rax, SYS_rt_sigprocmask);
-      syscall();
+/* ---------------------------------------------------------------------------------- */
 
-      // Bring stack back
-      add(rsp, 16);
+/*
+ * Call the dispatcher animation framework defined above, 
+ * while providing a quick termination via a longjmp:
+ */
+static 
+void X86DispatchCode( FEXCore::Core::CpuStateFrame *Frame )
+{
+  if (!setjmp(Frame->EmuContext)) {
+      uint32_t aligner MIE_ALIGN(16);
+      Frame->ReturningStackLocation = reinterpret_cast<uintptr_t>(&aligner);
 
-      jmp(r9);
-    }
-    else {
-      jmp(rax);
-    }
+      X86CallCoreDispatchCodeAsm(Frame);
   }
+}
 
-  {
-    // Pause handler
-    ThreadPauseHandlerAddress = getCurr<uint64_t>();
-    L(ThreadPauseHandler);
+__attribute__((noreturn))
+//static 
+void X86ThreadStopHandlerCode( FEXCore::Core::CpuStateFrame *Frame )
+{
+  longjmp(Frame->EmuContext,1);
+}
 
-    mov(rdi, reinterpret_cast<uintptr_t>(CTX));
-    mov(rsi, STATE);
-    mov(rax, reinterpret_cast<uint64_t>(SleepThread));
+/* ---------------------------------------------------------------------------------- */
 
-    call(rax);
+    __attribute__((noreturn))
+    static void WaitUntilWeHitATestCase(){assert(0);}
 
-    // XXX: Unsupported atm
-    PauseReturnInstruction = getCurr<uint64_t>();
-    ud2();
-  }
 
-  {
-    CallbackPtr = getCurr<JITCallback>();
 
-    push(rbx);
-    push(rbp);
-    push(r12);
-    push(r13);
-    push(r14);
-    push(r15);
-    sub(rsp, 8);
+static 
+void X86ThreadPauseHandlerAddressCode( FEXCore::Core::CpuStateFrame *Frame )
+#if 1
+{WaitUntilWeHitATestCase();}
+#else
+{
+  // Pause handler
+  SleepThread(Frame->Thread->CTX,Frame);
+  assert(false);
+}
+#endif
+  
+  
+static 
+void X86OverflowExceptionInstructionAddressCode( FEXCore::Core::CpuStateFrame *Frame )
+#if 1
+// Does not seem to be used yet???
+{WaitUntilWeHitATestCase();}
+#else
+#endif
+ /*
+    // ud2 = SIGILL
+    // int3 = SIGTRAP
+    // hlt = SIGSEGV
 
-    // First thing we need to move the thread state pointer back in to our register
-    mov(STATE, rdi);
-    // XXX: XMM?
+    mov(rax, reinterpret_cast<uint64_t>(&SynchronousFaultData));
+    add(byte [rax + offsetof(Dispatcher::SynchronousFaultDataStruct, FaultToTopAndGeneratedException)], 1);
+    mov(dword [rax + offsetof(Dispatcher::SynchronousFaultDataStruct, TrapNo)], X86State::X86_TRAPNO_OF);
+    mov(dword [rax + offsetof(Dispatcher::SynchronousFaultDataStruct, err_code)], 0);
+    mov(dword [rax + offsetof(Dispatcher::SynchronousFaultDataStruct, si_code)], 0x80);
 
+    hlt();
+  */
+  
+  
+static 
+void X86CallbackPtrCode( FEXCore::Core::CpuStateFrame *Frame, uint64_t RSI )
```

Comment:
These are used with thunking. So this definitely needs to keep working.

Reply:
How do I test this?